CanonicalLayouts/list-detail-compose/app/src/main/java/com/example/listdetailcompose/ui/ListDetailSample.kt

```diff
@@ -56,59 +74,84 @@ private val loremIpsum = """
         |Tempus quam pellentesque nec nam aliquam. Praesent semper feugiat nibh sed. Adipiscing elit duis tristique sollicitudin nibh sit. Netus et malesuada fames ac turpis egestas sed tempus urna. Quis varius quam quisque id diam vel quam. Urna duis convallis convallis tellus id interdum velit laoreet. Id eu nisl nunc mi ipsum. Fermentum dui faucibus in ornare. Nunc lobortis mattis aliquam faucibus. Vulputate mi sit amet mauris commodo quis. Porta nibh venenatis cras sed. Vitae tortor condimentum lacinia quis vel eros donec. Eu non diam phasellus vestibulum.
         """.trimMargin()
 private val sampleWords = listOf(
-    "Apple" to loremIpsum,
-    "Banana" to loremIpsum,
-    "Cherry" to loremIpsum,
-    "Date" to loremIpsum,
-    "Elderberry" to loremIpsum,
-    "Fig" to loremIpsum,
-    "Grape" to loremIpsum,
-    "Honeydew" to loremIpsum,
-).map { (word, definition) -> DefinedWord(word, definition) }
+    "Apple" to R.drawable.ic_food,
+    "Banana" to R.drawable.ic_no_food,
+    "Cherry" to R.drawable.ic_food,
+    "Date" to R.drawable.ic_no_food,
+    "Elderberry" to R.drawable.ic_food,
+    "Fig" to R.drawable.ic_no_food,
+    "Grape" to R.drawable.ic_food,
+    "Honeydew" to R.drawable.ic_no_food,
+).map { (word, icon) -> DefinedWord(word, icon) }
 
 private data class DefinedWord(
     val word: String,
-    val definition: String
+    @DrawableRes val icon: Int,
+    val definition: String = loremIpsum
 )
 
+@SuppressLint("UnusedContentLambdaTargetStateParameter")
 @OptIn(ExperimentalMaterial3AdaptiveApi::class)
 @Composable
 fun ListDetailSample() {
     var selectedWordIndex: Int? by rememberSaveable { mutableStateOf(null) }
     val navigator = rememberListDetailPaneScaffoldNavigator<Nothing>()
+    val isCompact =
+        currentWindowAdaptiveInfo().windowSizeClass.windowWidthSizeClass != WindowWidthSizeClass.EXPANDED
```

Comment:
This doesn't directly correspond to `isCompact` - since `isCompact` will be `true` even for a medium width. Looking at how it is used for the shared element logic, maybe the more specific condition that we care about is whether or not the `list` and `detail` are both visible at the same time?

Maybe we could pass through `isListVisible` and `isDetailVisible` to both the `ListContent` and `DetailContent` instead.

Reply:
I'll rename the parameter, I believe we care about the window space and the fact that the detail is visible to do the SharedElement Transition, am I missing something?

Reply:
I meant that the direct condition that we care about for the shared element transition is whether or not both the list and detail are showing at the same time.

With the default logic we have, that condition is exactly equivalent to `isExpanded`, but I thought it might be nicer to use the more direct condition which would be a bit more explicit for why it matters for the shared element: `isExpanded <==> isListAndDetailVisible`, so `isLessThanExpanded <==> !isListAndDetailVisible`

Reply:
Aaaaah, I see! In one of the Composables we check if it's less than expanded and if the detail is *not* visible, in the other we check for the size and if detail *is* visible. Is that correct?

Reply:
I was thinking to avoid passing the size down further, and have the two conditions be:

```
if (!isListAndDetailVisible && isDetailVisible)
```

```
if (!isListAndDetailVisible && isListVisible)
```

At the point, we don't care about the size directly, we care about whether both will be visible or not.